jetty-core/jetty-http2/jetty-http2-common/src/main/java/org/eclipse/jetty/http2/internal/HTTP2Flusher.java

```diff
@@ -163,7 +167,12 @@ protected Action process() throws Throwable
         try (AutoLock ignored = lock.lock())
         {
             if (terminated != null)
-                throw terminated;
+            {
+                 if (terminated instanceof HpackException.SessionException)
+                     terminated = new ClosedChannelException().initCause(terminated);
```

Comment:
Is the intention here is that next time `process()` is called, that we will throw the HE.SE wrapped as a CCE?

Maybe it would be clearer as
```java
             if (terminated instanceof HpackException.SessionException)
                 // Allow a SessionException to be processed once so that a GoAway can be sent
                 terminated = new ClosedChannelException().initCause(terminated);
             else if (terminated != null)
                 throw terminated;
```

This still feels a bit fragile.  If terminated is a HE.SE, then should we assert that pending contains a single entry that is a go-away?

Reply:
Done.